<?xml version="1.0"?>

<launch>
    <arg name="scenario_name" default="robocup_scenario"/>
    <arg name="traversability_mapping" default="$(optenv TRAVERSABILITY_MAPPING true)"/>
  
    <node pkg="vigir_worldmodel_server" type="worldmodel_server_node" name="worldmodel_main" output="screen" respawn="true">
        <remap from="scan_cloud" to="scan_cloud_trash" /> <!-- Disabled -->
        <!-- <remap from="scan_cloud_filtered" to="scan_matched_points2" />    -->
        <remap from="scan_cloud_filtered" to="colored_cloud" />

        <param name="aggregator_frames" value="/world /odom /base_link " />
        <param name="root_frame" value="/world" />

        <param name="required_frames" value="/base_link" />
        
        <param name="voxel_grid_size" value="0.03" />

        <param name="octomap_save_folder" value="$(find vigir_worldmodel_server)/maps" />
        <param name="use_external_octomap" value="false" />
        <param name="octomap_max_range" value="4.0" />

        <remap from="~/octomap_external_update" to="/move_group/octomap"/>
        <param name="publish_frames_as_poses" value="true" />

        <param name="periodic_octomap_save_folder" value="$(find robot_onboard_logging)/octomaps"/>
        <param name="periodic_octomap_save_period" value="30"/>

        <!--
        <remap from="get_distance_to_obstacle" to="/move_group/get_distance_to_obstacle" />
        <remap from="visualization_marker_array" to="worldmodel/visualization_marker_array" />
        -->
    </node>
</launch>
